spec/support/shared_validator_specs.rb

```diff
@@ -16,6 +16,31 @@ def alphanumeric_string(length)
   end
 end
 
+shared_examples "a property with data-types restriction" do
```

Comment:
I like this :+1: but it's not only "data-types", maybe rename it "... with validation" ?